src/st/cat/e_hellfire_beast.c

```diff
@@ -0,0 +1,1189 @@
+        prim->x1 = prim->x0;
+        prim->y0 = self->posY.i.hi;
+        if (self->facingLeft) {
+            prim->y0 -= 8;
+            prim->y1 = prim->y0 + 0x10;
+        } else {
+            prim->y0 += 8;
+            prim->y1 = prim->y0 - 0x10;
+        }
+
+        prim->x2 = self->posX.i.hi;
+        prim->y2 = prim->y0;
+        prim->x3 = prim->x2;
+        prim->y3 = prim->y1;
+
+        while (true) {
+            dx = posX - prim->x2;
+            dy = posY - prim->y2;
+            if (!self->ext.hellfireBeastThorsHammer.unk88) {
+                self->ext.hellfireBeastThorsHammer.unk88 = 4;
+                angle = ratan2(-dy, dx);
+                var_s2 = angle - var_s4;
+                if (var_s2 > 0x800) {
+                    var_s2 = 0x1000 - var_s2;
+                }
+                if (var_s2 < -0x800) {
+                    var_s2 = 0x1000 + var_s2;
+                }
+                var_s2 /= 4;
+                self->ext.hellfireBeastThorsHammer.unk86 = var_s2;
+            }
+
+            var_s4 += self->ext.hellfireBeastThorsHammer.unk86;
+            var_s4 &= 0xFFF;
+
+            nextPrim = prim->next;
+            if (nextPrim == NULL) {
+                self->step++;
+                break; // exits while loop
+            }
+
+            LOW(nextPrim->x0) = LOW(prim->x2);
+            LOW(nextPrim->x1) = LOW(prim->x3);
+            dx = (rcos(var_s4) * 0xC) >> 0xC;
+            dy = -((rsin(var_s4) * 0xC) >> 0xC);
+            nextPrim->x2 = nextPrim->x0 + dx;
+            nextPrim->y2 = nextPrim->y0 + dy;
+
+            var_s2 = var_s4 - 0x400;
+            var_s7 = 0x10;
+            dx = (var_s7 * rcos(var_s2)) >> 0xC;
+            dy = -(var_s7 * rsin(var_s2) >> 0xC);
+            nextPrim->x3 = nextPrim->x2 + dx;
+            nextPrim->y3 = nextPrim->y2 + dy;
+
+            nextPrim->u1 = var_s6;
+            nextPrim->u0 = var_s6;
+            nextPrim->u2 = nextPrim->u3 = nextPrim->u0 + 0xF;
+            var_s6 += 0x10;
+            if (var_s6 > 0x60) {
+                var_s6 = 0x10;
+            }
+
+            if (!(g_Timer % 2)) {
+                prevV0 = nextPrim->v0;
+                nextPrim->v0 = nextPrim->v2 = nextPrim->v1;
+                nextPrim->v1 = nextPrim->v3 = prevV0;
+            }
+            nextPrim->drawMode = DRAW_COLORS | DRAW_UNK02;
+            prim = nextPrim;
+            self->ext.hellfireBeastThorsHammer.unk88--;
+            i++;
+
+            if (i <= self->ext.hellfireBeastThorsHammer.unk8A) {
+                continue;
+            }
+
+            dx = nextPrim->x0 - self->posX.i.hi;
+            dx = abs(dx);
+            if (dx > 0x1FF) {
+                dx = 0x1FF;
+            }
+            self->hitboxOffX = -dx / 2;
+            self->hitboxWidth = dx / 2;
+            self->hitboxHeight = 8;
+            self->hitboxOffY = 8;
+
+            // FAKE! PSX requires some trickery here, was not able to resolve
+#ifdef VERSION_US
```

Comment:
Thanks to @Mc-muffin for the solve here - I couldn't work out a way to get rid of this on PSX side and make it clean like PSP unfortunately so have marked this as fake for now.